lib/src/geometry/line_segment.dart

```diff
@@ -0,0 +1,92 @@
+import 'line.dart';
+import '../../extensions.dart';
+
+class LineSegment {
```

Comment:
I think a doc here would be good specifically distinguishing Line and LineSegment to avoid confusion (and a similar one there).